### ISSUE TITLE
With few repo moved to pyproject.toml the sphinx docs generation needs modification

### ISSUE THREAD

**Issue by harshad16** [thoth-station/nepthys#98]
**Is your feature request related to a problem? Please describe.**

We build sphinx docs using the setup.py build_sphinx command, however as we have moved a few repositories to use pyproject.toml . we have to accommodate the new changes, so we can build sphinx docs using pyproject.toml.

```
+ pipenv run python3 setup.py build_sphinx
/home/user/.local/share/virtualenvs/analyzer-KKVhO2re/bin/python3: can't open file 'setup.py': [Errno 2] No such file or directory
```

**Additional context**
- [nepthys-27780480-zhgz2-nepthys.log](https://github.com/thoth-station/nepthys/files/9882506/nepthys-27780480-zhgz2-nepthys.log)
- https://github.com/sphinx-toolbox/sphinx-pyproject

**Comment by VannTen** [comment]
Looks like build_sphinx is deprecated anyway, yet there is not a standard remplacement. Python packaging, I guess :sob:

**Comment by VannTen** [comment]
More details:
Looks like direct setup.py invocation for anything are deprecated anyway.

see https://discuss.python.org/t/pep517s-definition-of-frontend-and-backend-is-unclear-to-me/16575/20?u=vannten

I see two options going forward:
- we invoke sphinx directly and format all of our project in the same way
- we use something like tox as a frontend handling the doc-building environment.
see https://github.com/tox-dev/tox/blob/legacy/tox.ini#L79-L86

@harshad16 @codificat any prefs ?

**Comment by harshad16** [comment]
we can try the first method, and use `sphinx-build` command instead  of `build_sphinx` from `setup.py`
then we can change this line:
https://github.com/thoth-station/nepthys/blob/b3ac82f366de8946d28104cfd4871daf32f7e84e/create_docs.sh#L44
to `PYTHONPATH=. pipenv run sphinx-build docs/source/ build/html`
it would fix the nepthys issues.
though setting version value https://github.com/thoth-station/python/blob/29050e8b2848f228caccd44cd6828c0077bde13f/docs/source/conf.py#L43 would be a problem.

**Comment by VannTen** [comment]
I think sphinx-pyproject might help with that, if we can pass the version from the pyproject.toml file.

**Comment by VannTen** [comment]
@Gkrumbach07 @harshad16 
From discussion in chat looks like there is some change which might affect how we do this ? Is that related to thoth-station/search#103 ?

**Comment by Gkrumbach07** [comment]
> Is that related to https://github.com/thoth-station/search/issues/103 ?

I don't think so. Chromatic only is used to evaluate UI components. It does not handle or use any docs gen. However when generating the docs, we will want them to be styled in the same way as the website. I [looked into](https://github.com/thoth-station/thoth-station.github.io/issues/245) using Gatsby to generate the docs, but that didn't seem feasible. However may be a way to add styling over the generated docs.

**Comment by harshad16** [comment]
As we might move to the gatsby-themed website, our requirements to generate HTML files might not be needed anymore. This would make further _feature_ development on nepthys low priority. 

we should fix the current bug, by moving components to pyproject style and making change to accommodate sphix build from pyproject.
